core/test/mpi/gtest/mpi_listener.cpp

```diff
@@ -383,5 +383,5 @@ int main(int argc, char** argv)
     listeners.Append(
         new GTestMPIListener::MPIWrapperPrinter(l, MPI_COMM_WORLD));
     int result = RUN_ALL_TESTS();
-    return 0;
+    return result;
```

Comment:
Is the reason of our test always succeed?

Reply:
In the old version yes. Now it gets correctly reported